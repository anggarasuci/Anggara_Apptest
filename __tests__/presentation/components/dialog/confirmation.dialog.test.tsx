import React from 'react';
import {render} from '@testing-library/react-native';
import ConfirmationDialog from '../../../../src/presentation/components/dialog/confirmation.dialog';

test('renders ConfirmationDialog correctly', () => {
  const onCancel = jest.fn();
  const onConfirm = jest.fn();

  const {toJSON} = render(
    <ConfirmationDialog
      label="Are you sure?"
      isVisible={true}
      onCancel={onCancel}
      onConfirm={onConfirm}
    />,
  );

  expect(toJSON()).toMatchSnapshot();
});
