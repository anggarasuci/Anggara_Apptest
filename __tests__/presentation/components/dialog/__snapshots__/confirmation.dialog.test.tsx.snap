// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ConfirmationDialog correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 5,
          "margin": 20,
          "padding": 20,
        }
      }
    >
      <Text>
        Are you sure?
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "marginTop": 20,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderColor": "gray",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Cancel
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderColor": "gray",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Confirm
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
